### ISSUE TITLE
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xbc in position 2: invalid start byte

### ISSUE THREAD

**Issue by chinaliuhan** [dbcli/mycli#842]
# UnicodeDecodeError: 'utf8'...

- system
```
MacOS 10.14.6
python3.8
```

- use this command install mycli
`brew update && brew install mycli`

- this is error message: 
```bash
~ ⌚ 1:38:22
$ mycli -u root
Traceback (most recent call last):
  File "/usr/local/bin/mycli", line 11, in <module>
    load_entry_point('mycli==1.20.1', 'console_scripts', 'mycli')()
  File "/usr/local/Cellar/mycli/1.20.1_3/libexec/lib/python3.8/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/Cellar/mycli/1.20.1_3/libexec/lib/python3.8/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/usr/local/Cellar/mycli/1.20.1_3/libexec/lib/python3.8/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/Cellar/mycli/1.20.1_3/libexec/lib/python3.8/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/Cellar/mycli/1.20.1_3/libexec/lib/python3.8/site-packages/mycli/main.py", line 1049, in cli
    mycli = MyCli(prompt=prompt, logfile=logfile,
  File "/usr/local/Cellar/mycli/1.20.1_3/libexec/lib/python3.8/site-packages/mycli/main.py", line 162, in __init__
    prompt_cnf = self.read_my_cnf_files(self.cnf_files, ['prompt'])['prompt']
  File "/usr/local/Cellar/mycli/1.20.1_3/libexec/lib/python3.8/site-packages/mycli/main.py", line 312, in read_my_cnf_files
    cnf = read_config_files(files, list_values=False)
  File "/usr/local/Cellar/mycli/1.20.1_3/libexec/lib/python3.8/site-packages/mycli/config.py", line 68, in read_config_files
    _config = read_config_file(_file, list_values=list_values)
  File "/usr/local/Cellar/mycli/1.20.1_3/libexec/lib/python3.8/site-packages/mycli/config.py", line 47, in read_config_file
    config = ConfigObj(f, interpolation=False, encoding='utf8',
  File "/usr/local/Cellar/mycli/1.20.1_3/libexec/lib/python3.8/site-packages/configobj.py", line 1229, in __init__
    self._load(infile, configspec)
  File "/usr/local/Cellar/mycli/1.20.1_3/libexec/lib/python3.8/site-packages/configobj.py", line 1287, in _load
    content = self._handle_bom(content)
  File "/usr/local/Cellar/mycli/1.20.1_3/libexec/lib/python3.8/site-packages/configobj.py", line 1437, in _handle_bom
    return self._decode(infile, self.encoding)
  File "/usr/local/Cellar/mycli/1.20.1_3/libexec/lib/python3.8/site-packages/configobj.py", line 1517, in _decode
    infile[i] = line.decode(encoding)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xbc in position 2: invalid start byte
```

please

**Comment by laixintao** [comment]
It seems there is an error of your `myclirc`, please check your `/etc/myclirc` and your `~/.myclirc`, see if there is any invalid utf-8 char in `prompt` section`.

**Comment by chinaliuhan** [comment]
> 看来您的错误`myclirc`，请检查您`/etc/myclirc`和您的`~/.myclirc`，看看在部分`prompt``中是否有任何无效的utf-8字符。

你好, 我试了, 没用.
thanks, i try it , but it's UNuseless.

**Comment by laixintao** [comment]
Can you show me your `~/.myclirc` and `/etc/myclirc`?

**Comment by chinaliuhan** [comment]
> 你能告诉我你的`~/.myclirc`和`/etc/myclirc`吗？

this is my `~/.myclirc` please.

```bash
$ cat .myclirc
# vi: ft=dosini
[main]

# Enables context sensitive auto-completion. If this is disabled the all
# possible completions will be listed.
smart_completion = True

# Multi-line mode allows breaking up the sql statements into multiple lines. If
# this is set to True, then the end of the statements must have a semi-colon.
# If this is set to False then sql statements can't be split into multiple
# lines. End of line (return) is considered as the end of the statement.
multi_line = False

# Destructive warning mode will alert you before executing a sql statement
# that may cause harm to the database such as "drop table", "drop database"
# or "shutdown".
destructive_warning = True

# log_file location.
log_file = ~/.mycli.log

# Default log level. Possible values: "CRITICAL", "ERROR", "WARNING", "INFO"
# and "DEBUG". "NONE" disables logging.
log_level = INFO

# Log every query and its results to a file. Enable this by uncommenting the
# line below.
# audit_log = ~/.mycli-audit.log

# Timing of sql statments and table rendering.
timing = True

# Table format. Possible values: ascii, double, github,
# psql, plain, simple, grid, fancy_grid, pipe, orgtbl, rst, mediawiki, html,
# latex, latex_booktabs, textile, moinmoin, jira, vertical, tsv, csv.
# Recommended: ascii
table_format = ascii

# Syntax coloring style. Possible values (many support the "-dark" suffix):
# manni, igor, xcode, vim, autumn, vs, rrt, native, perldoc, borland, tango, emacs,
# friendly, monokai, paraiso, colorful, murphy, bw, pastie, paraiso, trac, default,
# fruity.
# Screenshots at http://mycli.net/syntax
syntax_style = default

# Keybindings: Possible values: emacs, vi.
# Emacs mode: Ctrl-A is home, Ctrl-E is end. All emacs keybindings are available in the REPL.
# When Vi mode is enabled you can use modal editing features offered by Vi in the REPL.
key_bindings = emacs

# Enabling this option will show the suggestions in a wider menu. Thus more items are suggested.
wider_completion_menu = False

# MySQL prompt
# \D - The full current date
# \d - Database name
# \h - Hostname of the server
# \m - Minutes of the current time
# \n - Newline
# \P - AM/PM
# \p - Port
# \R - The current time, in 24-hour military time (0–23)
# \r - The current time, standard 12-hour time (1–12)
# \s - Seconds of the current time
# \t - Product type (Percona, MySQL, MariaDB)
# \u - Username
prompt = '\t \u@\h:\d> '
prompt_continuation = '-> '

# Skip intro info on startup and outro info on exit
#less_chatty = False
less_chatty = True

# Use alias from --login-path instead of host name in prompt
login_path_as_host = False

# Cause result sets to be displayed vertically if they are too wide for the current window,
# and using normal tabular format otherwise. (This applies to statements terminated by ; or \G.)
auto_vertical_output = False

# keyword casing preference. Possible values "lower", "upper", "auto"
keyword_casing = auto

# disabled pager on startup
enable_pager = True

# Custom colors for the completion menu, toolbar, etc.
[colors]
completion-menu.completion.current = 'bg:#ffffff #000000'
completion-menu.completion = 'bg:#008888 #ffffff'
completion-menu.meta.completion.current = 'bg:#44aaaa #000000'
completion-menu.meta.completion = 'bg:#448888 #ffffff'
completion-menu.multi-column-meta = 'bg:#aaffff #000000'
scrollbar.arrow = 'bg:#003333'
scrollbar = 'bg:#00aaaa'
selected = '#ffffff bg:#6666aa'
search = '#ffffff bg:#4444aa'
search.current = '#ffffff bg:#44aa44'
bottom-toolbar = 'bg:#222222 #aaaaaa'
bottom-toolbar.off = 'bg:#222222 #888888'
bottom-toolbar.on = 'bg:#222222 #ffffff'
search-toolbar = 'noinherit bold'
search-toolbar.text = 'nobold'
system-toolbar = 'noinherit bold'
arg-toolbar = 'noinherit bold'
arg-toolbar.text = 'nobold'
bottom-toolbar.transaction.valid = 'bg:#222222 #00ff5f bold'
bottom-toolbar.transaction.failed = 'bg:#222222 #ff005f bold'

# style classes for colored table output
output.header = "#00ff5f bold"
output.odd-row = ""
output.even-row = ""

# Favorite queries.
[favorite_queries]

# Use the -d option to reference a DSN.
# Special characters in passwords and other strings can be escaped with URL encoding.
[alias_dsn]
# example_dsn = mysql://[user[:password]@][host][:port][/dbname]



```

**Comment by j-bennet** [comment]
It can be any one of these files:

https://github.com/dbcli/mycli/blob/84dcba251fe10c3a8e8783ca997465256edbf475/mycli/main.py#L84-L87

@chinaliuhan it might be better if you attach the files (if present) instead of pasting the content.

**Comment by chinaliuhan** [comment]
> It can be any one of these files:
> 
> https://github.com/dbcli/mycli/blob/84dcba251fe10c3a8e8783ca997465256edbf475/mycli/main.py#L84-L87
> 
> @chinaliuhan it might be better if you attach the files (if present) instead of pasting the content.

thank you very much!
i remember  my `/usr/local/etc/my.cnf` has so many annotation.
like this : 
```sql

# 记录全局日志
log_output=file
general_log=ON
general_log_file=/tmp/db_general.log


# 慢查询日志
long_query_time = 0.3               # 超过多少秒记录
slow_query_log = ON                 # 慢查询日志开关
....
```

thank you told me, mysql's configuration file will effect the mycli.

**Comment by j-bennet** [comment]
@chinaliuhan It seems that `configobject` package developers fixed some Unicode errors in versions 5.0.1-5.0.6:

https://github.com/DiffSK/configobj/blob/master/CHANGES.rst

If you have an older version and are still experiencing the error, try updating configobj:

`pip install -U configobj`

**Comment by chinaliuhan** [comment]
> @chinaliuhan it might be better if you attach the files (if present) instead of pasting the content.

**Comment by chinaliuhan** [comment]
> @chinaliuhan It seems that `configobject` package developers fixed some Unicode errors in versions 5.0.1-5.0.6:
> 
> https://github.com/DiffSK/configobj/blob/master/CHANGES.rst
> 
> If you have an older version and are still experiencing the error, try updating configobj:
> 
> `pip install -U configobj`



thank you very much;

**Comment by chinaliuhan** [comment]
> @chinaliuhan It seems that `configobject` package developers fixed some Unicode errors in versions 5.0.1-5.0.6:
> 
> https://github.com/DiffSK/configobj/blob/master/CHANGES.rst
> 
> If you have an older version and are still experiencing the error, try updating configobj:
> 
> `pip install -U configobj`



thank you very much;
